bindings/matrix-sdk-crypto-js/scripts/epilogue.js

```diff
@@ -0,0 +1,81 @@
+// Initially, 'wasm' is set to a Proxy object which will synchronously load the webassembly and
+// replace 'wasm' with a reference to the exports from the wasm module.
+//
+// Ideally this will never get used because the application will call initAsync instead.
+wasm = new Proxy({}, {
+    get: (target, prop, receiver) => __initSync()[prop],
+});
+
+let inited = false;
+__initSync = function() {
+    if (inited) {
+        return;
+    }
+    if (initPromise) {
+        throw new Error("Asynchronous initialisation already in progress: cannot initialise synchronously");
+    }
+    const bytes = unbase64(require("./matrix_sdk_crypto_js_bg.wasm.js"));
+    const mod = new WebAssembly.Module(bytes);
+    const instance = new WebAssembly.Instance(mod, imports);
+    wasm = instance.exports;
+    wasm.__wbindgen_start();
+    inited = true;
+    return wasm;
+}
+
+let initPromise = null;
+
+/**
+ * Load the webassembly in the background, if it has not already been loaded.
+ *
+ * Returns a promise which will resolve once the other methods are ready.
+ *
+ * @returns {Promise<void>}
+ */
+module.exports.initAsync = function () {
+    if (inited) {
+        return Promise.resolve();
+    }
+    if (!initPromise) {
+        initPromise = Promise.resolve()
+            .then(() => require("./matrix_sdk_crypto_js_bg.wasm.js"))
+            .then(b64 => WebAssembly.instantiate(unbase64(b64), imports))
```

Comment:
Can we use `WebAssembly.instantiateStreaming` instead?

Also I didn't know `require` was a thing for a browser?

Reply:
> Can we use WebAssembly.instantiateStreaming instead?

It's hard to get hold of the wasm as a stream in an environment-independent way. For node.js we could use the `io` functions, but that doesn't work in a browser. In webpack we could maybe get the url for the wasm via https://webpack.js.org/api/module-variables/#importmetaurl but it's going to be very webpack-specific.

(we could transform the byte buffer into a stream, but I don't believe that offers any benefits?)

> Also I didn't know require was a thing for a browser?

It's not, but webpack and browserify implement it for you.